src/common/utilities/heatmap-utils.ts

```diff
@@ -1,39 +1,57 @@
+import { eachDayOfInterval } from 'date-fns';
 import { Logging } from "../../containers/habit-tracker/habits/Habits";
+import { formatDateLong } from "./date-utils";
 
-export interface HeatmapEntry {
-  month: number;
-  dayOfWeek: number;
-  logCount: number;
-  dayOfMonth: number;
+// Interface for the structure of a single heatmap data point
+interface HeatmapDataPoint {
+  x: number; // month (1-12)
+  y: number; // log count for that day
+  day: number; // The day of the month (1-31)
+}
+
+// Interface for the structure of the heatmap series
+export interface HeatmapSeries {
+  name: string; // Day of the week
+  data: HeatmapDataPoint[]; // Array of data points for the series
 }
 
 // Function to generate heatmap data from the logging data
 export function generateHeatmapData(
   habitLog: Logging,
   habitName: string,
   selectedYear: number
-): HeatmapEntry[] {
-  const habitLogData = habitLog[habitName];
-  const heatmapEntries: HeatmapEntry[] = [];
+): HeatmapSeries[] {
+  const logsForHabit = habitLog[habitName]; // Logs specific to the selected habit
+  const logsGroupedByWeekday = new Map<string, HeatmapDataPoint[]>(); // Map to group logs by the day of the week
+
+  // Dynamically get th number of days of the selected year
+  const daysInYear = eachDayOfInterval({
```

Comment:
there's a `getDaysInYear` method you can use. It is easier to use than this one.